do/droplets.go

```diff
@@ -144,6 +144,25 @@ func (i *instances) CurrentNodeName(hostname string) (types.NodeName, error) {
 // InstanceExistsByProviderID returns true if the instance for the given provider id still is running.
 // If false is returned with no error, the instance will be immediately deleted by the cloud controller manager.
 func (i *instances) InstanceExistsByProviderID(providerID string) (bool, error) {
+	id, err := dropletIDFromProviderID(providerID)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = dropletByID(context.TODO(), i.client, id)
+	if err == nil {
+		return true, nil
+	}
+
+	godoErr, ok := err.(*godo.ErrorResponse)
+	if !ok {
+		return false, fmt.Errorf("unexpected error type from godo: %T, %v", godoErr, err)
```

Comment:
`godoErr` will be nil in this case. Did you mean `err`?

Reply:
ahh yes, good catch